version: '3.8'

services:
  backend:
    build: ./backend
    volumes:
      - ./backend:/app
    ports:
      - "8080:8000"
    depends_on:
      db:
        condition: service_healthy
    environment:
      - DATABASE_URL=mysql+pymysql://root:password@db:3306/sutda
    networks:
      - app-network
    working_dir: /app
    command: uvicorn app.main:app --host 0.0.0.0 --port 8000 --reload


  front:
    build:
      context: ./front
    volumes:
      - ./front:/app
      - /app/node_modules
    ports:
      - "3000:3000"
    depends_on:
      - backend
    working_dir: /app
    command: npm run dev
    networks:
      - app-network

  db:
    image: mysql:8.0
    ports:
      - "3307:3306"
    environment:
      - MYSQL_ROOT_PASSWORD=password
      - MYSQL_DATABASE=sutda
      - MYSQL_USER=root
      - MYSQL_PASSWORD=password
    volumes:
      - mysql-data:/var/lib/mysql
    networks:
      - app-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-ppassword"]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  mysql-data:

networks:
  app-network:
    driver: bridge
